app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java

```diff
@@ -113,6 +139,19 @@ public void onLastVisibleItemPositionChanged(int position) {
     updateScrollDownFabVisibility(position);
   }
 
+  private void showSearchBox() {
+    hideSearchButton();
+  }
+
+  private void hideSearchButton() {
```

Comment:
This doesn't necessarily hide, it just toggles the current state.
It's a little tricky to understand from this code alone that changing the BOX boolean flag is effecting the box AND the FAB.